jenkins-scripts/docker/lib/_generic_linux_compilation_build.sh.bash

```diff
@@ -70,12 +70,27 @@ DELIM_BUILD_DEPS
   fi
 done
 
+if [[ -n ${ROS_DISTRO} ]]; then
```

Comment:
Change variable name accordingly here with the comment above. Also please include a comment on why we are using this in the _generic_linux_compilation_ (the fact of using the zenoh vendor for transport) since there are other scripts dealing with ROS.

Reply:
updated var and added comment [e25781b](https://github.com/gazebo-tooling/release-tools/pull/1311/commits/e25781be102940ddb378492d09b179cbb9e595d0). 

Actually if there's a better way to setup ROS env, let me know as well. I'm open to changing the implementation. I saw that there is some logic to setup ROS env in `boilerplate_prepare.sh` but I don't see it sourcing the setup script there. Maybe the alternative is to move the logic there.